bootloader/src/old_machine_bootloader_linux.c

```diff
@@ -154,19 +163,19 @@ int ping_island(int argc, char * argv[])
     if (!strcmp(osVersion, "")) {
         osVersion = executeCommand("cat /etc/system-release");
     }
-    printf("Os version: %s \n", osVersion);
+    printf("OS version: %s \n", osVersion);
 
     // Get all machine IP's
     int addrCount = 0;
     char** IPs = getIpAddresses(&addrCount);
     char* IPstring = "";
-    if (IPs != NULL){
+    if (NULL == IPs) {
```

Comment:
This was hard to pin point :/ I thought that you freed memory you shouldn't and combed trough getIpAddresses and concatenate :D